ddmrp/models/stock_warehouse_orderpoint.py

```diff
@@ -315,6 +337,19 @@ def _compute_dlt(self):
 
     _order = 'planning_priority_level asc, net_flow_position asc'
 
+    @api.model
+    def create(self, vals):
+        record = super(StockWarehouseOrderpoint, self).create(vals)
+        record._calc_adu()
+        return record
+
+    @api.multi
+    def write(self, vals):
+        super(StockWarehouseOrderpoint, self).write(vals)
+        if not self.env.context.get('__calc_adu'):
+            self._calc_adu()
+        return True
+
     @api.multi
     @api.onchange("red_zone_qty")
     def onchange_red_zone_qty(self):
```

Comment:
`red_zone_qty` is computed, but its computation doesn't trigger systematically the onchange.
In case the orderpoint `adu_calculation_method` is Fixed, and we change `fixed_adu`, the onchange is played correctly in the form view.
However, if we switch `adu_calculation_method` to Past, it's only when saving that `adu` is recomputed as is `red_zone_qty`. In this case the onchange isn't triggered and `product_min_qty` stays the same as it was when defined with fixed ADU.

The same behaviour applies to `onchange_green_zone_qty` at line 368.

Reply:
It's recommended to keep ADU computed daily and all the other parameters of the buffer the closest to real time possible. Knowing that, I wouldn't trigger the update of any value if ADU is detected to change (fixed adu or any other thing that related to ADU) but wait until the next daily update.